src/Initializer/time_stepping/LtsWeights/LtsWeights.cpp

```diff
@@ -66,26 +68,254 @@ class FaceSorter {
+
+    m_ncon = evaluateNumberOfConstraints();
+
+    // Compute cost
+    const double cost = computeGlobalCostOfClustering(m_clusterIds,
+                                                      m_cellCosts,
+                                                      m_rate,
+                                                      curWiggleFactor,
+                                                      m_details.globalMinTimeStep,
+                                                      MPI::mpi.comm());
+
+    if (auto it = mapNumberOfClustersToLowestCost.find(maxClusterId);
+        it == mapNumberOfClustersToLowestCost.end() || cost <= it->second) {
+      mapNumberOfClustersToBestWiggleFactor[maxClusterId] = curWiggleFactor;
+      mapNumberOfClustersToLowestCost[maxClusterId] = cost;
+    }
+  }
+
+  // Find best wiggle factor after merging of clusters
+  // We compare against cost of baseline (without wiggle, merging!)
+  int minAddmissibleNumberOfClusters = std::numeric_limits<int>::max();
+  if (ltsParameters->isAutoMergeUsed()) {
+    // When merging clusters, we want to find the minimum number of clusters with admissible performance.
+    bool foundAdmissibleMerge = false;
+    for (const auto& [noOfClusters, cost] : mapNumberOfClustersToLowestCost) {
+      if (cost <= maxAdmissibleCost) {
+        foundAdmissibleMerge = true;
+        minAddmissibleNumberOfClusters = std::min(minAddmissibleNumberOfClusters, noOfClusters);
+        logDebug(rank) << "Admissible. cluster:" << noOfClusters << ",cost" << cost
+                       << "with wiggle factor" << mapNumberOfClustersToBestWiggleFactor[noOfClusters];
+      } else {
+        logDebug(rank) << "Not admissible. cluster:" << noOfClusters << ",cost" << cost
+                       << "with wiggle factor" << mapNumberOfClustersToBestWiggleFactor[noOfClusters];
+      }
+    }
+    if (!foundAdmissibleMerge) {
+      logError() << "Found no admissible wiggle factor with cluster merge. Aborting.";
+    }
+  } else {
+    // Otherwise choose one with the smallest cost
+    minAddmissibleNumberOfClusters =
+        std::min_element(mapNumberOfClustersToLowestCost.begin(),
+                         mapNumberOfClustersToLowestCost.end(),
+                         [](const auto& a, const auto& b) { return a.second < b.second; })
+            ->first;
+  }
+
+  logInfo(rank) << "Enforcing maximum difference when finding best wiggle factor took"
+                << totalWiggleFactorReductions << "reductions.";
+
+  const auto bestWiggleFactor = mapNumberOfClustersToBestWiggleFactor[minAddmissibleNumberOfClusters];
+  const auto bestCostEstimate = mapNumberOfClustersToLowestCost[minAddmissibleNumberOfClusters];
+  logInfo(rank) << "The best wiggle factor after merging clusters is" << bestWiggleFactor
+                << "with cost" << bestCostEstimate;
+  logInfo(rank) << "Cost decreased" << (baselineCost - bestCostEstimate) / baselineCost * 100
```

Comment:
due to automerge, the cost can be increased.
(not it will say cost decrease by - xxx).